catalog/format/iceberg/src/main/java/org/projectnessie/catalog/formats/iceberg/nessie/NessieModelIceberg.java

```diff
@@ -654,10 +652,22 @@ public static NessieTableSnapshot icebergTableSnapshotToNessie(
               currentSnapshot.manifests(); // TODO
             });
 
-    for (IcebergSnapshotLogEntry logEntry : iceberg.snapshotLog()) {
-      // TODO ??
-      logEntry.snapshotId();
-      logEntry.timestampMs();
+    for (var icebergSnapshotLogEntry : iceberg.snapshotLog()) {
+      var snapId = icebergSnapshotLogEntry.snapshotId();
+      if (snapId == iceberg.currentSnapshotId()) {
+        continue;
+      }
+      if (previous == null) {
```

Comment:
Is `previous` ever non-null? IJ shows all non-test call paths leading to `null` :thinking:

Reply:
Those calls come from `importIcebergTable`... So, essentially if someone commits via Nessie API, each `NessieTableSnapshot` (after importing) will have duplicate "implicit" snapshot lists. WDYT?

Reply:
Yes - unfortunate.

Reply:
why keep `previous` if it is always `null` in production call paths?